### PR TITLE
feat(Clerk Upgrades): Upgrades to Clerk v5 [v5 is currently in beta]

### PR DESCRIPTION
**Pull request by JacobMGEvans**
- Upgrades required changes in Middleware                   
- Import changes for server side functions 
- UserProfile requires a path
https://github.com/clerk/javascript/blob/main/packages/nextjs/CHANGELOG.md

### CODE REVIEW

File: src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export async function generateMetadata(props: { params: { locale: string } }) {
 
 const UserProfilePage = () => (
   <div className="my-6 -ml-16">
-    <UserProfile />
+    <UserProfile path="/dashboard/user-profile" />
```
Comment (ixartz) [inline src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx:17]: How about adding path in SignIn and SignUp component? https://github.com/ixartz/Next-js-Boilerplate/blob/main/src/app/%5Blocale%5D/(auth)/(center)/sign-in/%5B%5B...sign-in%5D%5D/page.tsx#L16 and https://github.com/ixartz/Next-js-Boilerplate/blob/main/src/app/%5Blocale%5D/(auth)/(center)/sign-up/%5B%5B...sign-up%5D%5D/page.tsx

Based on the CHANGELOG example: https://github.com/clerk/javascript/blob/main/packages/nextjs/CHANGELOG.md#major-changes

**Review by ixartz — COMMENTED** [review]
Super existed about Clerk v5! Love the new import `nextjs/server`, love the new `createRouteMatcher`. It'll make the middleware much more easier to use.

File: src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx:17]: Yeah let me double check they are required on those as well. It was surfaced that their should be some sort of TS error or something being required for `UserProfile`

**Comment by JacobMGEvans** [comment]
> love the new createRouteMatcher.
We got that out of `experimental` with the latest 😄

File: src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx:17]: Yep its needed on all them, nice catch 😆

Comment (ixartz) [reply src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx:17]: Awesome, you commit the changes.

**Comment by ixartz** [comment]
Thank you so much for making the PR and the changes.

File: src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply src/app/[locale]/(auth)/dashboard/user-profile/[[...user-profile]]/page.tsx:17]: I did 😄

**Comment by JacobMGEvans** [comment]
> Thank you so much for making the PR and the changes.

Happy to help Clerk Community friends 😃

**Comment by ixartz** [comment]
@JacobMGEvans Thank you for making the change, I suppose we should wait until the version 5 is officially released, even I'm extremely excited.

**Comment by JacobMGEvans** [comment]
> @JacobMGEvans Thank you for making the change, I suppose we should wait until the version 5 is officially released, even I'm extremely excited.
Totally cool with that, we have a Beta cutting soon. I can push onto here when it does

**Comment by JacobMGEvans** [comment]
> > @JacobMGEvans Thank you for making the change, I suppose we should wait until the version 5 is officially released, even I'm extremely excited.
> 
> Totally cool with that, we have a Beta cutting soon. I can push onto here when it does

Additionally bumped to v5 Beta

**Comment by RayedB** [comment]
It seems like 5.2.1 is the latest version available now

**Comment by ixartz** [comment]
The boilerplate is now using Clerk v5 🎉

Thank you so much @JacobMGEvans for creating this PR during the beta, the upgrade is much more smoother.

**Pull request closed without merging**